src/mpl/include/mpl_base.h

```diff
@@ -155,4 +155,15 @@ typedef bool _Bool;
 #define MPL_ROUND_UP_ALIGN(a, alignment) (((a) + ((alignment) - 1)) & (~((alignment) - 1)))
 #define MPL_ROUND_DOWN_ALIGN(a, alignment) ((a) & (~((alignment) - 1)))
 
+#define MPL_CAST(T, val) CAST_##T((val))
+#ifdef NDEBUG
+#define MPL_CAST_int(val) ((int) (val))
+#define MPL_CAST_Aint(val) ((MPI_Aint) (val))
+#else
+#define MPL_CAST_int(val) \
+    (((val) > INT_MAX || ((val) < 0 && (val) < INT_MIN)) ? (assert(0), 0) : (int) (val))
```

Comment:
I think just `(val) > INT_MAX || (val) < INT_MIN` is sufficient.

Reply:
The `(val) < 0` case is to handle unsigned `val` which the compiler will internally convert `INT_MIN` to a unsigned format which makes `val < INT_MIN` always true for an unsigned `val`.

Reply:
Basically skipping the check with `INT_MIN` if `val` is unsigned.

Reply:
When `val` is `unsigned`, `(val) < INT_MIN` is always false